org.librarysimplified.audiobook.views/src/main/java/org/librarysimplified/audiobook/views/PlayerFragment.kt

```diff
@@ -55,27 +54,41 @@ import java.util.concurrent.TimeUnit
  * interface. An exception will be raised if this is not the case.
  */
 
-class PlayerFragment : Fragment() {
+class PlayerFragment(
+  private val listener: PlayerFragmentListenerType,
+  private val player: PlayerType,
+  private val book: PlayerAudioBookType,
+  private val scheduledExecutorService: ScheduledExecutorService,
+  private val sleepTimer: PlayerSleepTimerType
+) : Fragment() {
 
   companion object {
 
     const val parametersKey = "org.librarysimplified.audiobook.views.PlayerFragment.parameters"
 
     @JvmStatic
-    fun newInstance(parameters: PlayerFragmentParameters): PlayerFragment {
+    fun newInstance(
+      parameters: PlayerFragmentParameters,
+      listener: PlayerFragmentListenerType,
+      player: PlayerType,
+      book: PlayerAudioBookType,
+      scheduledExecutorService: ScheduledExecutorService,
+      sleepTimer: PlayerSleepTimerType
+    ): PlayerFragment {
       val args = Bundle()
       args.putSerializable(this.parametersKey, parameters)
-      val fragment = PlayerFragment()
+      val fragment = PlayerFragment(listener, player, book, scheduledExecutorService, sleepTimer)
       fragment.arguments = args
       return fragment
     }
   }
 
-  private lateinit var listener: PlayerFragmentListenerType
-  private lateinit var player: PlayerType
-  private lateinit var book: PlayerAudioBookType
-  private lateinit var executor: ScheduledExecutorService
-  private lateinit var sleepTimer: PlayerSleepTimerType
+  // private lateinit var listener: PlayerFragmentListenerType
+  // private lateinit var player: PlayerType
+  // private lateinit var book: PlayerAudioBookType
+  // private lateinit var executor: ScheduledExecutorService
+  // private lateinit var sleepTimer: PlayerSleepTimerType
+
```

Comment:
I see these are now in the constructor. Is it safe to go ahead and delete these now?

Reply:
Yep, good catch.